src/project-generators/react-basic/index.ts

```diff
@@ -1,220 +1,123 @@
     variation: ReactComponentStylingFlavors.CSSModules,
   })
 
-  const routingComponentGenerator = createRouterComponentGenerator()
-
-  // Step 1: Building the folder structure (rooted in dist by default) for the React project
-  const componentsFolder: GeneratedFolder = {
-    name: 'components',
-    files: [],
-    subFolders: [],
-  }
-
-  const pagesFolder: GeneratedFolder = {
-    name: 'pages',
-    files: [],
-    subFolders: [],
-  }
-
-  const staticFolder: GeneratedFolder = {
-    name: 'static',
-    files: [],
-    subFolders: [],
-  }
-
-  const srcFolder: GeneratedFolder = {
-    name: 'src',
-    files: [],
-    subFolders: [componentsFolder, pagesFolder, staticFolder],
-  }
-
-  const distFolder: GeneratedFolder = {
-    name: options.distPath || DEFAULT_OUTPUT_FOLDER,
-    files: [],
-    subFolders: [srcFolder],
-  }
-
-  // Step 2: Initialization with project specific mappings and of other data structures
   reactGenerator.addMapping(reactProjectMapping)
   if (options.customMapping) {
     reactGenerator.addMapping(options.customMapping)
   }
 
-  let collectedDependencies: Record<string, string> = {}
-  const { components = {}, root } = uidl
-  const { states } = root.content
-  const stateDefinitions = root.stateDefinitions
-
-  const result = {
-    outputFolder: distFolder,
-    assetsPath: 'src' + ASSETS_PREFIX,
-  }
+  return reactGenerator
+}
 
-  if (!states || !stateDefinitions) {
-    return result
-  }
+export default async (uidl: ProjectUIDL, options: ProjectGeneratorOptions = {}) => {
+  // Step 0: Initialize the component generator
+  const reactGenerator = initGenerator(options)
 
-  const routerDefinitions = stateDefinitions.router
-  if (!routerDefinitions) {
-    return result
-  }
+  const { components = {}, root } = uidl
+  const { states = [] } = root.content
+
+  const stateDefinitions = root.stateDefinitions || {}
+  const routerDefinitions = stateDefinitions.router || null
+
+  // Step 1: The first level stateBranches (the pages) transformation in react components is started
+  const pagePromises = states.map(async (stateBranch) => {
+    const pageParams: PageFactoryParams = {
+      componentGenerator: reactGenerator,
+      stateBranch,
+      routerDefinitions,
+      componentOptions: {
+        assetsPrefix: ASSETS_PREFIX,
+        localDependenciesPrefix: LOCAL_DEPENDENCIES_PREFIX,
+      },
+    }
+    return createPageFile(pageParams)
+  })
 
-  // Step 3: Global settings are transformed into the root html file and the manifest file for PWA support
-  if (uidl.globals.manifest) {
-    const manifestJSON = createManifestJSON(uidl.globals.manifest, uidl.name, ASSETS_PREFIX)
-    const manifestFile: GeneratedFile = {
-      name: 'manifest',
-      extension: '.json',
-      content: JSON.stringify(manifestJSON, null, 2),
+  // Step 2: The components generation process is started
+  const componentPromises = Object.keys(components).map(async (componentName) => {
```

Comment:
should the callback function passed on to the `map` be `async`? Looks like it should not.